src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1274,36 +1274,56 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         // ComCtl scrollbars, but it's certainly close enough.
         const auto scrollbarSize = GetSystemMetricsForDpi(SM_CXVSCROLL, dpi);
 
-        float width = gsl::narrow<float>(cols * fFontWidth);
+        double width = cols * fFontWidth;
+        double height = rows * fFontHeight;
 
         // Reserve additional space if scrollbar is intended to be visible
         if (settings.ScrollState() == ScrollbarState::Visible)
         {
             width += scrollbarSize;
         }
-
-        const float height = gsl::narrow<float>(rows * fFontHeight);
-
-        return { width, height };
+        
+        // Reserve space for padding
+        const auto paddingArr = _ParsePadding(settings.Padding());
+        switch (paddingArr.size())
+        {
+            case 1:
+                width += paddingArr[0] * 2;
+                height += paddingArr[0] * 2;
+                break;
+            case 2:
+                width += paddingArr[0] * 2;
+                height += paddingArr[1] * 2;
+                break;
+            // No case for 3 padding values, since it's not a norm to provide just Left, Top & Right padding values leaving out Bottom
+            case 4:
+                width += paddingArr[0] + paddingArr[2];
+                height += paddingArr[1] + paddingArr[3];
+                break;
+            default:
+                break;
+        }
+        
+        return { static_cast<float>(width), static_cast<float>(height) };
     }
 
     // Method Description:
-    // - Create XAML Thickness object based on padding props provided.
-    //   Used for controlling the TermControl XAML Grid container's Padding prop.
+    // - Parse the given padding props to a std::vector.
     // Arguments:
     // - padding: 2D padding values
     //      Single Double value provides uniform padding
     //      Two Double values provide isometric horizontal & vertical padding
     //      Four Double values provide independent padding for 4 sides of the bounding rectangle
     // Return Value:
-    // - Windows::UI::Xaml::Thickness object
-    Windows::UI::Xaml::Thickness TermControl::_ParseThicknessFromPadding(const hstring padding)
+    // - A std::vector object
+    // - If parsed successfully the vector will have the same size as the number of padding values.
+    // - If parsing failed the vector will be empty.
+    std::vector<double> TermControl::_ParsePadding(const hstring padding)

```

Comment:
This function probably does not belong in this file but I don't know where should I put it.